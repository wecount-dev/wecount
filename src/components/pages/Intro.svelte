<script lang="ts">
  import supabase from '../../lib/db';
  import {_} from 'svelte-i18n';
  import Button from '../uis/Button.svelte';

  const signOut = () => supabase.auth.signOut();

  function toggleTheme() {
    window.document.body.classList.toggle('dark-mode');
  }
</script>

<p>{$_('intro.title')}</p>
<Button on:click={signOut}>Sign Out</Button>
<Button on:click={toggleTheme}>Toggle Theme</Button>
