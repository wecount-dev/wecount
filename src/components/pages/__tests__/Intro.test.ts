// @ts-ignore
import Intro from '../Intro.svelte';
import {render} from '@testing-library/svelte';

it('should render without error', () => {
  const testngLib = render(Intro);

  expect(testngLib.container).toMatchSnapshot();
});
