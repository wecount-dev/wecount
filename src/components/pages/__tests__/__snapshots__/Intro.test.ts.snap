// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without error 1`] = `
<body>
  <div>
    <p>
      Intro
    </p>
     
    <button
      class="svelte-1eav0tq"
    >
      Sign Out
    </button>
     
    <button
      class="svelte-1eav0tq"
    >
      Toggle Theme
    </button>
  </div>
</body>
`;
